lib/Listener/CalDavEventListener.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2019 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Talk\Listener;
+
+use OCA\DAV\CalDAV\TimezoneService;
+use OCA\DAV\Events\CalendarObjectCreatedEvent;
+use OCA\DAV\Events\CalendarObjectUpdatedEvent;
+use OCA\Talk\Exceptions\RoomNotFoundException;
+use OCA\Talk\Manager;
+use OCA\Talk\Room;
+use OCA\Talk\Service\RoomService;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\IUserManager;
+use Psr\Log\LoggerInterface;
+use Sabre\VObject\Property\ICalendar\Date;
+use Sabre\VObject\Property\ICalendar\DateTime;
+use Sabre\VObject\Reader;
+
+/** @template-implements IEventListener<CalendarObjectCreatedEvent|CalendarObjectUpdatedEvent> */
+class CalDavEventListener implements IEventListener {
+
+	public function __construct(
+		private Manager $manager,
+		private RoomService $roomService,
+		private LoggerInterface $logger,
+		private IUserManager $userManager,
+		private TimezoneService $timezoneService,
+		private string $userId,
+	) {
+
+	}
+
+	public function handle(Event $event): void {
+		if (!$event instanceof CalendarObjectCreatedEvent && !$event instanceof CalendarObjectUpdatedEvent) {
+			return;
+		}
+
+		$calData = $event->getObjectData()['calendardata'] ?? null;
+		if (!$calData) {
+			return;
+		}
+
+		if (!str_contains($calData, 'LOCATION:')) {
+			$this->logger->debug('No location for the even, skipping.');
+			return;
+		}
+
+		$vobject = Reader::read($calData);
+		$vevent = $vobject->VEVENT;
+		// Check if the location is set and if the location string contains a call url
+		$location = $vevent->LOCATION->getValue();
+		if ($location === null || !str_contains($location, '/call/')) {
+			$this->logger->debug('No location for the event or event is not call link, skipping.');
+			return;
+		}
+
+		// Check if room exists and check if user is part of room
+		$roomToken = array_reverse(explode('/', $location))[0];
+		try {
+			$room = $this->manager->getRoomByToken($roomToken, $this->userId);
```

Comment:
`getRoomByToken` does not force a membership. It just loads it when it exists. `getRoomForUserByToken` throws `RoomNotFoundException` when the user is not a member, so we should use that:
```suggestion
			$room = $this->manager->getRoomForUserByToken($roomToken, $this->userId);
```

but it still returns open and listable rooms, so afterwards we should get the membership (will not run a query when the user was found before):
```php
$participant = $this->participantService->getParticipant($room, $userId, false);
```
that throws ParticipantNotFoundException => return

Should we limit this to moderators, same as with the "schedule a meeting"?